File: rust/src/models/coupon_line_item.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CouponLineItem {
    pub coupon_id: String,

    pub name: String,

    pub total: i32,
}

impl CouponLineItem {
    pub fn new(coupon_id: String, name: String, total: i32) -> Self {
        Self {
            coupon_id,
            name,
            total,
        }
    }
}

File: typescript/src/models/couponLineItem.ts
// this file is @generated

export interface CouponLineItem {
  couponId: string;

  name: string;

  total: number;
}

export const CouponLineItemSerializer = {
  _fromJsonObject(object: any): CouponLineItem {
    return {
      couponId: object["coupon_id"],
      name: object["name"],
      total: object["total"],
    };
  },

  _toJsonObject(self: CouponLineItem): any {
    return {
      coupon_id: self.couponId,
      name: self.name,
      total: self.total,
    };
  },
};
